lib/spack/spack/main.py

```diff
@@ -420,6 +419,24 @@ def make_argument_parser(**kwargs):
     parser.add_argument(
         '--print-shell-vars', action='store',
         help="print info needed by setup-env.[c]sh")
+    parser.add_argument(
+        '-u', '--upstream', dest='upstream',
+        action='store', default=None,
+        help='target specified upstream'
+    )
```

Comment:
`--upstream` and `--install-root` both set `spack.store.install_root`

Reply:
This is being kept (temporarily) for backwards compatibility for users who were already using the original `--upstream` option at our site.

Reply:
Since we're approaching a merge on this PR I've removed this option. The --install-root [name] command accomplishes the same thing.